napari/layers/surface/surface.py

```diff
@@ -190,6 +195,27 @@ def dtype(self):
     def data(self):
         return (self.vertices, self.faces, self.vertex_values)
 
+    @data.setter
+    def data(self, data):
+        if len(data) not in (2, 3):
+            raise ValueError(
+                trans._(
+                    'Surface data tuple must be 2 or 3, specifying'
+                    'vertices, faces, and optionally vertex values,'
+                    'instead got length {data_length}.',
+                    deferred=True,
+                    data_length=len(data),
+                )
+            )
+        self._vertices = data[0]
+        self._faces = data[1]
+        if len(data) == 3:
+            self._vertex_values = data[2]
+        else:
+            self._vertex_values = np.ones(len(self._vertices))
+
+        self._update_dims()
```

Comment:
Should we emit a data event in the data setter? I think we do this in the other layers.